plutus-core/testlib/PlutusCore/Generators/Internal/Denotation.hs

```diff
@@ -51,45 +54,54 @@ data DenotationContextMember term res =
 --   2. a bound variable of functional type with the result being @integer@
 --   3. the 'AddInteger' 'Builtin' or any other 'Builtin' which returns an @integer@.
 newtype DenotationContext term = DenotationContext
-    { unDenotationContext :: DMap (AsKnownType term) (Compose [] (DenotationContextMember term))
+    { unDenotationContext :: DMap TypeRep (Compose [] (DenotationContextMember term))
```

Comment:
This is the gist of the PR: replacing `AsKnownType` with `TypeRep` everywhere. Allows us to get rid of that ridiculous `unsafeCoerce` that we had, support more builtins and refactoring builtins stuff becomes much easier, 'cause these tests no longer get in the way as much.